### PR TITLE
feat (OONI Run v2): update preference screens to match new designs

### PR DESCRIPTION
**Pull request by aanorbel**
Fixes https://github.com/ooni/ooni.org/issues/1516

## Proposed Changes

  - Updated Preference view.
  - Remove cell separators.

|.|.|.|
|-|-|-|
| Simulator Screenshot - iPhone 15 Pro Max - 2024-06-17 at 12 16 22| Simulator Screenshot - iPhone 15 Pro Max - 2024-06-17 at 12 16 27 | Simulator Screenshot - iPhone 15 Pro Max - 2024-06-17 at 12 16 31 |
| Simulator Screenshot - iPhone 15 Pro Max - 2024-06-17 at 12 16 36 | Simulator Screenshot - iPhone 15 Pro Max - 2024-06-17 at 12 16 42 | Simulator Screenshot - iPhone 15 Pro Max - 2024-06-17 at 12 16 58 |
| Simulator Screenshot - iPhone 15 Pro Max - 2024-06-17 at 12 17 04 |. |.|

### CODE REVIEW

File: ooniprobe/Utility/SettingsUtility.m [hunk 1]
```diff
@@ -28,25 +29,25 @@ + (NSArray*)getSettingsForCategory:(NSString*)categoryName{
...
         return nil;
+    }
 }
 
 + (NSString*)getTypeForSetting:(NSString*)setting{
-    if ([setting isEqualToString:@"experimental"] || [setting isEqualToString:@"long_running_tests_in_foreground"])
+    if ([setting isEqualToString:@"experimental"] || [setting isEqualToString:@"long_running_tests_in_foreground"]) {
         return @"bool";
-    else if ([setting isEqualToString:@"website_categories"] || [[TestUtility getTestOptionTypes] containsObject:setting])
+    } else if ([setting isEqualToString:@"website_categories"]) {
```
Comment (bassosimone) [inline ooniprobe/Utility/SettingsUtility.m:41]: What is the impact of deleting this clause?

File: ooniprobe/View/Settings/SettingsTableViewController.m [hunk 2]
```diff
@@ -63,8 +65,9 @@ -(void)viewWillDisappear:(BOOL)animated{
         [testSuite.testList removeAllObjects];
         [testSuite getTestList];
     }
-    if (testSuite != nil || [[TestUtility getTestOptionTypes] containsObject:category])
+    if (testSuite != nil) {
```
Comment (bassosimone) [inline ooniprobe/View/Settings/SettingsTableViewController.m:68]: Same question as above: what is the impact of removing this clause?

File: ooniprobe/View/Settings/SettingsTableViewController.m [hunk 3]
```diff
@@ -113,9 +116,6 @@ - (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(N
         else {
             cell = [tableView dequeueReusableCellWithIdentifier:@"Cell" forIndexPath:indexPath];
         }
-        if ([[TestUtility getTestOptionTypes] containsObject:current]){
-            cell.imageView.image = [UIImage imageNamed:current];
-        }
```
Comment (bassosimone) [inline ooniprobe/View/Settings/SettingsTableViewController.m:118]: Can you help me understand what was the purpose of this check and of the code executed if true?

File: ooniprobe/View/Settings/SettingsTableViewController.m [hunk 4]
```diff
@@ -132,13 +132,7 @@ - (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(N
         }
         else
             cell = [tableView dequeueReusableCellWithIdentifier:@"Cell" forIndexPath:indexPath];
-        if ([[TestUtility getTestOptionTypes] containsObject:current]){
-            if ([NSLocale characterDirectionForLanguage:[NSLocale preferredLanguages][0]] == NSLocaleLanguageDirectionRightToLeft) {
-                cell.imageView.image = [self imageWithImage:[UIImage imageNamed:current] convertToSize:CGSizeMake(32, 32)];
-            } else {
-                cell.imageView.image = [UIImage imageNamed:current];
-            }
-        }
```
Comment (bassosimone) [inline ooniprobe/View/Settings/SettingsTableViewController.m:141]: Can you help me understand what was the purpose of this check and of the code executed if true?

**Review by bassosimone — COMMENTED** [review]
I mainly have questions and curiosities about the impact of changes

File: ooniprobe/View/Settings/SettingsTableViewController.m (hunk 4 above)
Comment (aanorbel) [reply ooniprobe/View/Settings/SettingsTableViewController.m:141]: This was previously used to to set the icon for test categories based on the language. RTL languages would have the icons on the left and LTR languages on the other side.

File: ooniprobe/Utility/SettingsUtility.m (hunk 1 above)
Comment (aanorbel) [reply ooniprobe/Utility/SettingsUtility.m:41]: `[TestUtility getTestOptionTypes]` is no longer one of the expected settings type.

Comment (bassosimone) [reply ooniprobe/Utility/SettingsUtility.m:41]: ACK

File: ooniprobe/View/Settings/SettingsTableViewController.m (hunk 3 above)
Comment (bassosimone) [reply ooniprobe/View/Settings/SettingsTableViewController.m:118]: Got it

File: ooniprobe/View/Settings/SettingsTableViewController.m (hunk 4 above)
Comment (bassosimone) [reply ooniprobe/View/Settings/SettingsTableViewController.m:141]: Understood!

File: ooniprobe/View/Settings/SettingsTableViewController.m (hunk 2 above)
Comment (aanorbel) [reply ooniprobe/View/Settings/SettingsTableViewController.m:68]: [TestUtility getTestOptionTypes] is no longer one of the expected settings type.

**Pull request merged**